lib/api/BackbeatAPI.js

```diff
@@ -705,6 +867,23 @@ class BackbeatAPI {
                 error: errors.MalformedPOSTRequest.customizeDescription(errMsg),
             };
         }
+        // Bucket, Key, VersionId, StorageClass
+        // filter duplicate version ids if any
+        const uniqueObjs = {};
+        if (process.env.CI === 'true') {
+            // ignore filtering for tests
```

Comment:
Why do tests need to ignore filtering?

Reply:
failure keys rely on valid bucket, key, version ids. I wasn't fully sure how the test mock metadata worked, but the version id's in the failure keys for tests use the same mock metadata. I'd like to eventually adjust the tests, and left comment here https://github.com/scality/backbeat/pull/416/files/df1c6700ef3c549deebdfe3a6b0b6a7a2dd8cbdf#diff-8c901706f0bfac9a0fd4f1d1205a4924R267

I've mainly relied on manual testing for these fixes since the bug reported mainly has to do with user usage cases or timing

Reply:
OK, maybe worth commenting with a TODO in the code as it does not come obvious.